src/app/index/update/inserter.rs

```diff
@@ -91,34 +87,26 @@ impl Inserter {
 	}
 
 	fn flush_directories(&mut self) {
-		if self
-			.db
-			.connect()
-			.and_then(|connection| {
-				diesel::insert_into(directories::table)
-					.values(&self.new_directories)
-					.execute(&*connection) // TODO https://github.com/diesel-rs/diesel/issues/1822
-					.map_err(Error::new)
-			})
-			.is_err()
-		{
+		let res = self.db.connect().and_then(|connection| {
+			diesel::insert_into(directories::table)
+				.values(&self.new_directories)
+				.execute(&*connection) // TODO https://github.com/diesel-rs/diesel/issues/1822
```

Comment:
The linked issue is closed, but I haven't looked over it closely, can we now do `.execute(&connection)`?